packages/app/src/debug/DebugFailedTest.vue

```diff
@@ -1,33 +1,150 @@
 <template>
   <div
     data-cy="test-row"
-    class="flex flex-row items-center gap-x-2.5 non-italic text-base text-gray-700 font-normal"
+    class="flex flex-row h-12 items-center non-italic text-base text-gray-700 font-normal"
   >
-    <div
-      v-for="(titlePart, index) in props.failedTestResult.titleParts"
+    <SolidStatusIcon
+      size="16"
+      status="failed"
+      data-cy="failed-icon"
+      class="isolate"
+    />
+    <template
+      v-for="titlePart, index in failedTestData.mappedTitleParts"
       :key="`${titlePart}-${index}`"
-      class="flex flex-row items-center gap-x-2.5"
       :data-cy="`titleParts-${index}`"
     >
       <IconChevronRightSmall
-        v-if="index !== 0"
-        data-cy="right-chevron"
+        v-if="index !== 0 && titlePart.type !== 'LAST-1'"
+        :data-cy="`titleParts-${index}-chevron`"
         size="8"
         stroke-color="gray-200"
         fill-color="gray-200"
+        class="shrink-0"
+        :class="titlePart.type === 'MIDDLE' ? 'hidden lg:block' : titlePart.type === 'ELLIPSIS' ? 'lg:hidden' : ''"
       />
-      <span>
-        {{ titlePart }}
+      <span
+        :data-cy="`titleParts-${index}-title`"
+        :class="titlePart.type === 'ELLIPSIS' ? 'px-2.5 shrink-0 lg:hidden' :
+          titlePart.type === 'MIDDLE' ? 'truncate px-2.5 hidden lg:block' :
+          titlePart.type === 'LAST-1' ? 'shrink-0 whitespace-pre' :
+          titlePart.type === 'LAST-0' ? 'pl-2.5 truncate' : 'px-2.5 truncate'"
+      >
+        {{ titlePart.title }}
       </span>
+    </template>
+    <div
+      v-if="!props.expandable"
+      data-cy="debug-artifacts"
+      class="flex flex-grow space-x-4.5 opacity-0 pr-18px justify-end test-row-artifacts"
```

Comment:
```suggestion
      class="flex flex-grow space-x-4.5 opacity-0 px-18px justify-end test-row-artifacts"
```
This also needs some left padding to keep the artifact links away from test titles that take up all the space.